src/main/scala/com/gu/newsletterlistcleanse/sqs/AwsSQSSend.scala

```diff
@@ -47,26 +28,19 @@ object AwsSQSSend {
     (sqsClient, queueUrl)
   }
 
-  def sendAsync(sqsClient:AmazonSQSAsync ,queueUrl: String, payload: SinglePayload): Future[SendMessageResult] = {
-    AwsAsync(sqsClient.sendMessageAsync, new SendMessageRequest(queueUrl, payload.value))
-  }
-
-  def sendBatchAsync(sqsClient:AmazonSQSAsync ,queueUrl: String, batchPayload: BatchPayload): Future[SendMessageBatchResult] = {
-    val batchRequest: java.util.List[SendMessageBatchRequestEntry] = batchPayload.value.map({case(id, payload) =>
-      new SendMessageBatchRequestEntry(id, payload)}
-    ).toList.asJava
+  private def sendMessage(queueName: QueueName, payload: Payload): SendMessageResult = {
+    val (sqsClient: AmazonSQS, queueUrl: String) = buildSqsClient(queueName)
 
-    AwsAsync.apply[SendMessageBatchRequest, SendMessageBatchResult](
-      sqsClient.sendMessageBatchAsync,
-      new SendMessageBatchRequest(queueUrl, batchRequest)
-    )
+    sqsClient.sendMessage(queueUrl, payload.value)
   }
 
-  case class QueueName(value: String) extends AnyVal
 
-  sealed trait Payload
+  def sendCleanseList(queueName: QueueName, cleanseList: CleanseList): SendMessageResult = {
```

Comment:
my only comment here is that this SQS send class should be a generic code for sending arbitrary messages (so that someone else can steal it later for another app) and all domain specific types e.g. CleanseList should go in your higher level code.  Baascially separation of concerns to make the code easier to understand.

Reply:
That seems sound. I was trying to put in some sort of type safety around what gets converted to JSON, but it makes sense to put that in the respective lambdas rather than in a shared module. Ta!